infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -249,12 +249,19 @@ private void fillHole(CorfuPayloadMsg<FillHoleRequest> msg, ChannelHandlerContex
         }
     }
 
+    /**
+     * Perform a prefix trim.
+     * Here the token is not used to perform the trim as the epoch at which the checkpoint was completed
+     * might be old. Hence, we use the msg epoch to perform the trim. This should be safe provided that the
+     * trim is performed only on the token provided by the CheckpointWriter which ensures that the checkpoint
+     * was persisted. Using any other address to perform a trim can cause data loss.
+     */
     @ServerHandler(type = CorfuMsgType.PREFIX_TRIM)
     private void prefixTrim(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext ctx,
                             IServerRouter r) {
         try {
             TrimRequest req = msg.getPayload();
-            batchWriter.prefixTrim(req.getAddress());
+            batchWriter.prefixTrim(new Token(msg.getEpoch(), req.getAddress().getSequence()));
```

Comment:
The token is not persisted and this new code doesn't use the epoch. I suggest you drop the Token usage instead of transforming it.

Reply:
Done.